site/content/docs/developing/overrides/yamlpatch.md

```diff
@@ -0,0 +1,123 @@
+    Please use with caution!
+
+Copilot generates CloudFormation templates using configuration specified in the [manifest](../../manifest/overview.en.md). 
+However, not all CloudFormation properties are configurable in the manifest. 
+For example, you might want to configure the [`Ulimits`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-ulimit.html) 
+for your workload container, but the property is not exposed in manifests.
+
+Overrides with `yamlpatch` or `cdk` allows you to add, delete, or replace _any_ property or resource in a CloudFormation template.
+
+## When should I use YAML Patch over CDK overrides?
+
+Both options are a "break the glass" mechanism to access and configure functionality that is not surfaced by Copilot [manifests](../../manifest/overview.en.md).  
+
+We recommend using YAML patch over the [AWS Cloud Development Kit (CDK) overrides](./cdk.md) if 1) you do not want to have a dependency
+on any other tooling and framework (such as [Node.js](https://nodejs.org) and the [CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html)), 
+or 2) you have to write only a handful modifications. 
+
+## How to get started
+
+You can extend your CloudFormation template with YAML patches by running the `copilot [noun] override` command.
+For example, you can run `copilot svc override` to update the template of a Load Balanced Web Service.
+The command will generate a sample `cfn.patches.yml` file under the `copilot/[name]/override` directory. 
+
+## How does it work?
+
+The syntax of `cfn.patches.yml` conforms to [RFC6902: JSON Patch](https://www.rfc-editor.org/rfc/rfc6902). Currently, 
+the CLI supports three operations: `add`, `remove`, and `replace`. Here is a sample `cfn.patches.yml` file:
+
+```yaml
+- op: add
+  path: /
+  value:
+    Mappings:
+      ContainerSettings:
+        test: { Cpu: 256, Mem: 512 }
+        prod: { Cpu: 1024, Mem: 1024}
+- op: remove
+  path: /Resources/TaskRole
+- op: replace
+  path: /Resources/TaskDefinition/Properties/ContainerDefinitions/1/Essential
+  value: false
+- op: add
+  path: /Resources/Service/Properties/ServiceConnectConfiguration/Services/0/ClientAliases/-
+  value:
+    Port: !Ref TargetPort
+    DnsName: yamlpatchiscool
+```
+
+Each patch is applied sequentially to the CloudFormation template. The resulting template becomes the target of the next patch. 
+Evaluation continues until all patches are successfully applied or an error is encountered.
+
+### Path evaluation
+
+The `path` field of a patch conforms to the [RFC6901: JSON Pointer](https://www.rfc-editor.org/rfc/rfc6901) syntax. 
+
+- Each `path` value is separated by the `/` character and evaluation stops once the target CloudFormation property is reached.
+- If the target path is an array, the reference token must either be:
+    - characters comprised of digits starting at 0.
+    - exactly the single character `-` when the operation is `add` to append to the array.
```

Comment:
it seems like you can only use `-`  in the last element. maybe we should let users know

Reply:
paging @dannyrandall, my undestanding is that `-` is basically only useful for `add` (`-` refers to a non-existing last element)

Reply:
that's correct! `-` is only valid as the last element in the pointer for an `add` operation